src/Microsoft.PowerShell.Commands.Management/commands/management/Computer.cs

```diff
@@ -1503,12 +1503,28 @@ private void DoRenameComputerWsman(string computer, string computerName, string
 
                         if (!InternalTestHooks.TestRenameComputer)
                         {
-                            CimMethodResult result = cimSession.InvokeMethod(
-                                ComputerWMIHelper.CimOperatingSystemNamespace,
-                                cimInstance,
-                                "Rename",
-                                methodParameters,
-                                operationOptions);
+                            CimMethodResult result = null;
+
+                            if (isLocalhost)
+                            {
+                                var computerSystem = cimSession.EnumerateInstances(ComputerWMIHelper.CimOperatingSystemNamespace, ComputerWMIHelper.WMI_Class_OperatingSystem);
+
+                                result = cimSession.InvokeMethod(
+                                    ComputerWMIHelper.CimOperatingSystemNamespace,
+                                    computerSystem.FirstOrDefault(),
+                                    ComputerWMIHelper.CimOperatingSystemShutdownMethod,
```

Comment:
Is this correct?  This is calling Shutdown, but the else case is calling rename?

Reply:
Remove this piece of code as no change is needed in `Rename-Computer`